From 95de7859529e1bafc5f6e16034f16767af4fde45 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@freescale.com>
Date: Thu, 26 Mar 2015 18:27:51 +0800
Subject: [PATCH 0249/1221] MLK-10480-1 ARM: dts: imx7d-12x12-arm2: add adc
 support

Add ADC1 support for imx7d-12x12-arm2 board.

Signed-off-by: Haibo Chen <haibo.chen@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-arm2.dts |   12 ++++++++++++
 1 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
index cf2ba93..e888294 100644
--- a/arch/arm/boot/dts/imx7d-12x12-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-arm2.dts
@@ -107,6 +107,13 @@
 			enable-active-high;
 		};
 
+		reg_vref_1v8: regulator@2 {
+			compatible = "regulator-fixed";
+			regulator-name = "vref-1v8";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+		};
+
 		reg_can1_3v3: can1-3v3 {
 			compatible = "regulator-fixed";
 			regulator-name = "can1-3v3";
@@ -129,6 +136,11 @@
 	};
 };
 
+&adc1 {
+	vref-supply = <&reg_vref_1v8>;
+	status = "okay";
+};
+
 &cpu0 {
 	arm-supply = <&sw1a_reg>;
 };
-- 
1.7.5.4

